### PR TITLE
Hydraspec Package(With preserved commits)

### PR DESCRIPTION
**Pull request by shravandoda**
Please use `pip install -e git+https://github.com/ShravanDoda/hydra-python-core@develop#egg=hydra_python_core` to install.
I've preserved the history of commits as well.

### CODE REVIEW

**Comment by de-sh** [comment]
@ShravanDoda is it important to have all that history? I guess it would be better to make a patch, also I think it's not necessary to have the .egg-info directory within the repo right?

**Comment by shravandoda** [comment]
> @ShravanDoda is it important to have all that history? I guess it would be better to make a patch, also I think it's not necessary to have the .egg-info directory within the repo right?

As far as history is concerned, I think it's necessary to preserve it because if we're going to separate the package into it's own repo, we might need to reference the commits in the future. @Mec-iS will agree I think. .egg-info directory is not needed. I'll remove those

**Comment by de-sh** [comment]
> As far as history is concerned, I think it's necessary to preserve it because if we're going to separate the package into it's own repo, we might need to reference the commits in the future. 

Yeah, but we gotta keep only the history of the files within hydrus/hydraspec right? For this we can use a git patch...

**Comment by de-sh** [comment]
Sorry, didn't see that, I was trying out patch and it wasn't working properly... My mistake 😅

**Comment by Mec-iS** [comment]
name of the library/package should be `hydra-python-core` and in Python naming system `hydra_python_core`

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from setuptools import setup, find_packages
```
Comment (de-sh) [inline setup.py]: Would have been better if you kept with the use of defining dependencies within the requirements.txt file. You could find the [sample setup.py here](https://github.com/HTTP-APIs/python-hydra-agent/blob/master/setup.py)

Comment (shravandoda) [reply setup.py]: Which dependencies? I don't think hydra-python-core has any.

Comment (de-sh) [reply setup.py]: I meant to point to the use of find_packages() 😅

**Comment by Mec-iS** [comment]
Please write an extensive `README.md` to explain how the library is packaged and how to install it.

**Comment by shravandoda** [comment]
@Mec-iS, the link for installation in README.md references HTTP-APIs' hydra-python-core repo(for future)

**Review by vddesai1871 — COMMENTED** [review]
Looks good!
We should also move  `doc_writer` test here

**Comment by de-sh** [comment]
>Looks good!
>We should also move doc_writer test here

@ShravanDoda please [review my PR](https://github.com/ShravanDoda/hydra-python-core/pull/1) :smile:

**Comment by shravandoda** [comment]
@de-sh the tests are failing due to an import error. You're tests are using 'hydrus.hydraspec.doc_writer'. Kindly change these to 'hydra_python_core.doc_writer'. Morever, please preserve the commits as well(if there are any)

**Comment by de-sh** [comment]
Sorry for that @ShravanDoda :sweat_smile:, I have made the necessary changes.

**Comment by de-sh** [comment]
Sorry for that @ShravanDoda :sweat_smile:, I have made the necessary changes.

**Comment by shravandoda** [comment]
@de-sh the previous commits are not present. Please check again.

**Comment by shravandoda** [comment]
@vddesai1871 test_doc_writer.py has been moved into the repo.

**Review by de-sh — COMMENTED** [review]
So, just curious, but why is the directory structure with the python snake-case? `hydra_python_core` instead of `hydra-python-core` within this repo?

**Comment by shravandoda** [comment]
> So, just curious, but why is the directory structure with the python snake-case? `hydra_python_core` instead of `hydra-python-core` within this repo?

Python variables cannot have '-' in their names. The 'hydra_python_core' is a python package.

**Comment by shravandoda** [comment]
@Mec-iS what's the status here?

**Comment by Mec-iS** [comment]
@ShravanDoda Please align your changes to current `master`/`develop` and open a new PR

**Comment by Mec-iS** [comment]
PS. The way for porting history is by using `git subtree` --> https://stackoverflow.com/a/2989613

[UPDATE]
How to proceed:
* remove the `hydraspec` and `samples` directory
* write a new `setup.py` for the new `python-hydra-core` repo
* make hydrus to install the new repo at setup time
* modify the imports as `from python_hydra_core  ...`

**Pull request closed without merging**